Tools/px_uploader.py

```diff
@@ -287,13 +287,13 @@ def __ackSyncWindow(self, count):
             if (len(data) != count):
                 raise RuntimeError("Ack Window %i not %i " % (len(data), count))
             for i in range(0, len(data), 2):
-                if chr(data[i]) != self.INSYNC:
+                if chr(data[i]).encode() != self.INSYNC:
```

Comment:
`encode` uses utf-8 by default, so I'd rather use `bytes([data[i]])` (in python2 it's still `chr(data[i])` though, but we can consider dropping python 2 support).

Reply:
I'm not super good at python so I will let you use your best judgement on that

Reply:
I'd create a separate method then we can keep the compatibility for now:
```
def to_byte(data):
  if runningPython3:
    return bytes([data])
  return chr(data)
```

Reply:
My two cents are: remove the python2 compatibility stuff and raise an exception if the script is run with Python 2. Python2 is EOL and the shebang clearly states python3.

Reply:
Super weird, I made the changes but it still fails. I even added a type check...
```
Type data[0]: <class 'bytes'>
Type self.INSYNC: <class 'bytes'>

ERROR: unexpected 18 instead of INSYNC
```

Reply:
Can you show the diff? You forgot the additional brackets: `bytes([data[i]])`